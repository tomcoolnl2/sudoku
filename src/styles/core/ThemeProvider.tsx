
import { FC } from 'react'
import { useSelector } from 'react-redux'
import { ThemeProvider } from 'styled-components'
import { lightTheme, darkTheme, GlobalStyles } from './'
import { StoreReducer } from '../../redux'


export interface DarkThemeProviderState {
    darkModeEnabled: boolean
}

export const DarkThemeProvider: FC = ({ children }) => {
	const { darkModeEnabled } = useSelector<StoreReducer, DarkThemeProviderState>(state => state.settings)
	const theme = darkModeEnabled ? darkTheme : lightTheme
	return (
		<ThemeProvider theme={theme}>
			<GlobalStyles />
			{children}
		</ThemeProvider>
	)
}