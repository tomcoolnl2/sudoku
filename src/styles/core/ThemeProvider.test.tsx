
import { render } from '@testing-library/react'
import { withRedux } from '../../test'
import { DarkThemeProvider } from './ThemeProvider'


describe('ThemeProvider', () => {
	
	it('should match it\'s snapshot', () => {
		const { container } = render(withRedux(<DarkThemeProvider />))
		expect(container.firstChild).toMatchSnapshot()
	})
})