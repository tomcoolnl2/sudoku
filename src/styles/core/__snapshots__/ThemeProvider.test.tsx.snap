// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should match it's snapshot 1`] = `null`;
